docs/mp/config/02_MP_config_sources.adoc

```diff
@@ -0,0 +1,230 @@
+
+Custom Config Sources are loaded using the Java ServiceLoader pattern and can be registered by providing a file named `org.eclipse.microprofile.config.spi.ConfigSource` in the `/META-INF/services/` folder.
+
+You need to use the following methods and implement the interface `org.eclipse.microprofile.config.spi.ConfigSource` to create a custom Config Source:
+
+* `String getName()`
+* `Map<String, String> getProperties()`
+* `String getValue(String key)`
+* `getOrdinal()`
+
+=== Example of a Custom Config Source
+
+[source,java]
+----
+public class CustomConfigSource implements ConfigSource {
+			String NAME = "Gh1468ConfigSource";
+			PRIORITY = 200; // Default for MP is 100
+			Map<String, String> PROPERTIES = mapOf("app.greeting", "Hi");
+
+
+	@Override
+	public String getName() { 
+		return NAME; <1>
+	}
+
+	@Override
+	public Map<String, String> getProperties() { 
+		return PROPERTIES; <2>
+	}
+
+	@Override
+	public String getValue(String key) {
+		return PROPERTIES.get(key); <3>
+	}
+
+	@Override
+	public int getOrdinal() {
+		return PRIORITY; <4>
+	}
+
+}
+----
+
+<1> Returns the name of the Config Source to use for logging or analysis of configured values.
+<2> Returns the properties in this Config Source as a map.
+<3> Returns the value stored in the propertyName of this Config Source.
+<4> Returns the ordinal based on the set order for this Config Source.
+
+
+== Creating MicroProfile Config Sources for Manual Setup of Config
+
+You can use the following methods to create MicroProfile Config Sources to manually set up the Config from `org.eclipse.microprofile.config.spi.ConfigProviderResolver#getBuilder()`:
+
+|===
+|Method |Description
+
+|`systemProperties()}`   |System properties config source.
+
+|`environmentVariables()}`   |Environment variables config source.
+
+|`create(java.nio.file.Path)}`   |Loads a properties file from file system. +
+To load the properties file from file system with custom name, use `create(String, java.nio.file.Path)}`.
+
+|`create(java.util.Map)}`   |Creates an in-memory source from map. +
+To create an in-memory source from map with custom name, use `create(String, java.util.Map)}`.
+
+|`create(java.util.Properties)}`   |Creates an in-memory source from properties. +
+To create an in-memory source from properties with custom name, use `create(String, java.util.Properties)}`.
+
+
+|===
+
+===  Config Source Creation based on System Properties
+
+The Config Source queries the properties each time `org.eclipse.microprofile.config.spi.ConfigSource#getValue(String)` is called.
+
+[source,java]
+----
+ConfigSource systemProperties() {
+        return new MpSystemPropertiesSource();
+----
+
+=== Config Source Creation based on Environment Variables
+
+The Config Source replaces the properties with environment variables as defined in MicroProfile Config specification.
+
+[source,java]
+----
+ConfigSource environmentVariables() { <1>
+        return new MpEnvironmentVariablesSource();
+----

```

Comment:
This is not correct. Helidon classes do not have public constructors. You need to use the factory methods in `MpConfigSources`